### PR TITLE
Added RAdamScheduleFree support | updated codecarbon to run test_trainer.py properly

### PR DESCRIPTION
**Pull request by nhamanasu**
# What does this PR do?

Integrates meta's https://github.com/facebookresearch/schedule_free for RAdam.
This PR greatly followed the previous PR to introduce schedulefree https://github.com/huggingface/transformers/pull/30079

## More details
- Added support for using RAdamScheduleFree through Trainer API
- Refined doc for trainer (clarified that we recommend to use constant LR scheduler with Schedule-Free family)
- Added test case for RAdamScheduleFree & confirmed the test passes
  - Environment: RTX 6000Ada (no distributed training, fp32)
  - 113 passed, 49 skipped, 52 warnings in 133.76s (0:02:13)

## Backgrounds
I'm the developer who recently released the RAdamScheduleFree optimizer.
It can be imported via schedulefree library since version 1.4.

- [The author of Schedule-Free paper's announcement post](https://x.com/aaron_defazio/status/1863722349138219459)
- [My announcement and release blog (in Japanese)](https://x.com/hamanasu_nagisa/status/1866413964235821492)

RAdamScheduleFree eliminates:
- Learning rate scheduler and terminal steps
- Warmup steps

## Bug Fixes
- Related to [this codecarbon PR](https://github.com/mlco2/codecarbon/pull/391), I found we need to update codecarbon to run test_trainer.py properly in Python 3.11 or above. In current main branch, `"codecarbon==1.2.0"` was set, so I rewrote this to `"codecarbon>=2.8.1"`. It properly worked.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@muellerzr @younesbelkada @pacman100

### CODE REVIEW

**Comment by nhamanasu** [comment]
CircleCI shows "[tests_generate](https://app.circleci.com/pipelines/github/huggingface/transformers/113356/workflows/1e152e04-8f26-496a-9925-549ac7435e88/jobs/1514539?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary) - Failed", but I think it's not related to my commits...

**Comment by nhamanasu** [comment]
I decided to split this PR into several smaller PRs to address each concern separately:
- [first PR] https://github.com/huggingface/transformers/pull/35243

I'll close this PR and submit a new one focusinga on RAdamScheduleFree once all concerns are resolved.
